src/Contracts/Contracts.ts

```diff
@@ -23,4 +24,10 @@ export enum KubeImage {
     zeroData = "zeroData",
     zeroResults = "zeroResults",
     zeroWorkloads = "zeroWorkloads"
+}
+
+export interface IImageService {
+    hasImageDetails(listImages: Array<string>): Promise<{ [key: string]: boolean } | undefined>;

```

Comment:
why undefined here and in next method?

Reply:
The return type of getDataAsync as defined in ContributionService is Promise<TData | undefined>, hence this is needed
![image](https://user-images.githubusercontent.com/12389221/54510474-6f5ddd80-4973-11e9-844b-eeb1064a1342.png)